### PR TITLE
Use + Provide Conan Packages

### PR DESCRIPTION
**Pull request by johannes-wolf**
This adds a `conanfile.py` to simfil (also in the conan-index fork: https://github.com/conan-io/conan-center-index/compare/master...Klebert-Engineering:conan-center-index:simfil-0.1.0), which allows it to be used via conan (`conan editable add simfil` or `conan create . --build=missing -s compiler.cppstd=20`).

Also…
- Replaced use of `stx` with `fmtlib` (stx::split got copied over to get rid of the stx dependency)
- Adds CodeQL for code scanning
- Adds a Conan `conan create` CI run to create and test the package

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "fPIC": [True, False],
+        "with_json": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "with_json": True,
+    }
+
+    # Sources are located in the same place as this recipe, copy them to the recipe
+    exports_sources = "CMakeLists.txt", "include/*", "src/*"
```
Comment (josephbirkner) [inline conanfile.py:30]: Shouldn't there be square brackets around the values?

Comment (johannes-wolf) [reply conanfile.py:30]: Python's wonderful tuple syntax.

**Comment by johannes-wolf** [comment]
To be able to release to conan.io, we need to first do a simfil "release". See https://github.com/conan-io/conan-center-index/issues/22625#issuecomment-1923316189.

**Comment by josephbirkner** [comment]
Looks good! Can you update the description? Is the plan that conanfile.py in conan center can read the dependencies from conanfile.txt here?

**Comment by johannes-wolf** [comment]
> Looks good! Can you update the description? Is the plan that conanfile.py in conan center can read the dependencies from conanfile.txt here?

No, it can not read the conanfile.txt. We should drop the conanfile from this repo.
You can still use this repo without conan (see deps.cmake). I am not sure, but I guess we have to have two conanfiles. The recipe (py) can not (and I think it should not) read this repo's files.

**Comment by johannes-wolf** [comment]
TODOs
- [ ] Copy over `conanfile.py` from the conan-center back to this repo and replace `conanfile.txt` (this allows for `conan editable add .`)
- [ ] Document conan package development in the readme file

**Comment by johannes-wolf** [comment]
We should wait for the `sfl/1.2.3` PR getting merged to conan-index because using `editable` mode is currently not working because of that.

**Comment by josephbirkner** [comment]
Very cool

**Review by josephbirkner — APPROVED** [review]

**Comment by johannes-wolf** [comment]
Let's merge this and do follow up fixes for the conanfile in a separate PR.

**Pull request merged**